packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -583,7 +583,7 @@ export class DefaultCrudRepository<
       const relName = def.relations[r].name;
       if (relName in data) {
         throw new Error(
-          `Navigational properties are not allowed in model data (model "${this.entityClass.modelName}" property "${relName}")`,
+          `Navigational properties are not allowed in model data (model "${this.entityClass.modelName}" property "${relName}"). Please remove it or make sure the relation name is not the same as the property name in belongsTo relation.`,
```

Comment:
Should it be `belongsTo` or `hasMany`? 
Like `Customer` hasMany `orders`, IIUC the navigational property here is `orders`, which is the relation name for hasMany

Reply:
Yes, 
For hasMany, `orders` would only be the navigational property. When you create a customer, you would do `customerRepo.create({id: 1, name: 'customer'});` 

But for belongsTo, by how we defining the default relation name, the fk `customer_id` would have the default relation name as `customer_id`. When you do `orderRepo.create({id: 1, des: 'order', customer_id: 1})`, it recognizes the `customer_id` in the data as the **_relation name_** instead of the foreign key. That's why I state it clear that that the error might be caused by the nav property OR the naming issue here. 

Please let me know if you have any suggestions to make it more intuitive.

Reply:
@agnes512 👍 thank you got it. How about the following warning:

Navigational properties are not allowed in model data (model "${this.entityClass.modelName}" property "${relName}"), please remove it. 
For relation belongsTo, please make sure the relation name is not the same as the property name.